### DISCUSSION TITLE
What do you want to see in a new Dashboard resource?

### DISCUSSION THREAD

**Discussion by mbazhlekova** [newrelic/terraform-provider-newrelic#1480] [Ideas]
Hi all,

We're thinking of creating a new Dashboard resource for a future TF provider release that is more user friendly and can support newer dashboard features. Please use this space to share any ideas you have for improving the Dashboard resource. 

We're excited to hear from you!

**Comment by rdhar** [comment]
At the base level, supporting dashboard-specific metadata would be useful, such as: "tv-mode: show the dashboard name" and "tv-mode: cycle to the next page every 20 seconds". That should get rid of the current workaround of having to manually configure every single dashboard generated (which isn't feasible after a certain point).

Beyond that, I'm curious about the scope of this discussion. Particularly with regards to current existence of both `one_dashboard` and `one_dashboard_raw`. Do you mean something to replace both of these resources, or adding on top of them?

For example, the biggest drawback of dashboards for me is the difficulty of setting widgets to filter on their page. Current workarounds are far from ideal. Recent releases include a cryptic `filter_current_dashboard` boolean attribute but it [doesn't seem applicable as of yet](https://github.com/newrelic/terraform-provider-newrelic/issues/1413#issuecomment-944962081).

**Reply by kidk** [reply]
Great feedback @rdhar 

In general we would like to combine `one_dashboard` and `one_dashboard_raw`, or at least introduce a `widget_raw` to `one_dashboard`, but there are technical limitations with that.

**Reply by rdhar** [reply]
I wonder if it's worth considering support for raw JSON in the same format as you would get from any dashboard's "Copy JSON by clipboard" button?

This puts the configuration onus on the user while maintaining full compatibility with the rest of New Relic and its "Import dashboard" button via the UI. This also reduces development management of the dashboard resource since you're dealing with the JSON directly. Similar implementation to webhook `alert_channel` resource with [JSON payload](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/alert_channel#webhook).

Though it does add abstraction to the code, it's very flexible as well.

**Comment by rdhar** [comment]
Happy to confirm that [v2.38.0](https://github.com/newrelic/terraform-provider-newrelic/releases/tag/v2.38.0) has resolved the main `filter_current_dashboard = true` issue observed with `newrelic_one_dashboard`. Many thanks for shipping!

**Comment by cdobbyn** [comment]
Right now managing dashboards as a single resource works, but is difficult to manage for more complicated usages. I would like to see the ability to define pages as a separate resource, maybe even the widgets themselves.

Additionally being able to place some of the more unique widgets (like kubernetes nodes) would be nice.

**Reply by kidk** [reply]
Thanks for feedback @cdobbyn! Once we have a draft of a new resource we will post it here. Splitting up the resources is going to require API changes, so we will need to work with the dashboard team to make that happen. But we're hopeful :D

**Comment by FlareLine** [comment]
Just checking in with this one @kidk, not sure if this is currently being worked on, or if there is a beta available? Looks like the last activity was in 2022. 😁

**Reply by FlareLine** [reply]
Just looking into potentially picking up https://github.com/newrelic/terraform-provider-newrelic/pull/2531, but if NR is looking at rearching the APIs, it might be worth it to look into this discussion instead?

**Comment by abudri** [comment]
The ability to take existing new relic alarms / conditions and reference them by resource id or other and build out a dashboard of alarms from as many alarms as we like. AWS Cloudwatch dashboards have that ability to take an alarm and add it to a dashboard of your choosing. We've been able to Terraform that as well, which is ultimately the goal with a new relic dashboard of this kind containing alarms / alert conditions, to get an overall alarming view of all our alarms across services and environments, and not just be limited to the APM of a particular environment of a service for a given dashboard.
